src/main/java/com/liferay/ide/idea/language/tag/LiferayTaglibClassNameReferenceContributor.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.ide.idea.language.tag;
+
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiReferenceProvider;
+import com.intellij.psi.impl.source.resolve.reference.impl.providers.JavaClassReferenceProvider;
+import com.intellij.psi.xml.XmlAttribute;
+import com.intellij.psi.xml.XmlTag;
+import com.intellij.psi.xml.XmlToken;
+import com.intellij.psi.xml.XmlTokenType;
+
+import java.util.AbstractMap;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+/**
+ * @author Dominik Marks
+ */
+public class LiferayTaglibClassNameReferenceContributor extends AbstractLiferayTaglibReferenceContributor {
+
+	@Override
+	protected String[] getAttributeNames() {
+		return new String[] {
+			"assetCategoryClassName", "assetTagClassName", "className", "exception", "model", "modelResource",
+			"portletProviderClassName"
+		};
+	}
+
+	@Override
+	protected PsiReferenceProvider getPsiReferenceProvider() {
+		JavaClassReferenceProvider javaClassReferenceProvider = new JavaClassReferenceProvider();
+
+		javaClassReferenceProvider.setOption(JavaClassReferenceProvider.ADVANCED_RESOLVE, Boolean.TRUE);
+		javaClassReferenceProvider.setOption(JavaClassReferenceProvider.RESOLVE_QUALIFIED_CLASS_NAME, Boolean.TRUE);
+
+		return javaClassReferenceProvider;
+	}
+
+	@Override
+	protected boolean isSuitableXmlAttribute(XmlAttribute xmlAttribute) {
+		if (_containsTextOnly(xmlAttribute)) {
```

Comment:
There are 3 methods that are very simillar.  Here
Please two others.  

https://github.com/gamerson/liferay-intellij-plugin/pull/68/files#diff-ab34e2fbfc3e8aa2fc46e35723575bbcR44
https://github.com/gamerson/liferay-intellij-plugin/pull/68/files#diff-4d07cf42ce17f52c21eb722e408e6ce1R43

I'm going to merge this one, but @dmarks2  could you take a 2nd look and this and see if you think these 3 methods could be generalized into one abstract method to share the code better between the 3 implementations?  

Or feel free to do a new class that all 3 could call out to if you want to use composition instead of inheritance.

Reply:
@gamerson Thank you for your feedback. I will have a look at the methods and try to generalize them. I will send you a new pull request when finished.

Reply:
sent as https://github.com/gamerson/liferay-intellij-plugin/pull/72